src/checkers/inference/solver/MaxSat2TypeSolver.java

```diff
@@ -77,7 +77,11 @@ public InferenceSolution solve() {
         //if an exception occurs while creating a variable the id might be incremented
         //but the slot might not actually be recorded.  Therefore, nextId is NOT
         //the number of slots but the maximum you might encounter.
-        final int totalVars = slotManager.getNumberOfSlots();
+        // TODO: this is a workaround as currently when serialize existential constraint we lost the real existential
+        // TODO: variable id and create "fake" id stored in existentialToPotentialVar map.
+        // TODO: thus here the value of totalVars is the real slots number stored in slotManager, and plus the 
+        // TODO: "fake" slots number stored in existentialToPotentialVar
+        final int totalVars = slotManager.getNumberOfSlots() + serializer.getExistentialToPotentialVar().size();
```

Comment:
Also here, shouldn't the number of slots be multiplied with the number of bits per slot?